src/Sylius/Bundle/ApiBundle/Resources/config/services.xml

```diff
@@ -164,5 +164,23 @@
             <argument type="service" id="Sylius\Bundle\ApiBundle\DataProvider\ChannelAwareItemDataProvider.inner" />
             <argument type="service" id="sylius.context.channel" />
         </service>
+
+        <service id="Sylius\Bundle\ApiBundle\Converter\ItemIriToIdentifierConverter">
+            <argument type="service" id="api_platform.router" />
+        </service>
+
+        <service id="Sylius\Bundle\ApiBundle\Serializer\CommandFieldItemIriToIdentifierDenormalizer">
+            <argument type="service" id="serializer.normalizer.object" />
+            <argument type="service" id="Sylius\Bundle\ApiBundle\Converter\ItemIriToIdentifierConverter" />
+            <argument type="service" id="sylius.api.data_transformer.command_aware_input_data_transformer" />
+            <argument type="tagged_locator" tag="sylius.api.command_field_item_iri_to_identifier_storage" />
+            <tag name="serializer.normalizer" />
+        </service>
+
+        <service id="AddProductReviewCommandFieldItemIriToIdentifierStorage" class="Sylius\Bundle\ApiBundle\Converter\CommandFieldItemIriToIdentifierStorage">
+            <argument type="string">Sylius\Bundle\ApiBundle\Command\AddProductReview</argument>
+            <argument type="string">product</argument>
+            <tag name="sylius.api.command_field_item_iri_to_identifier_storage" />
```

Comment:
This is my main concern about this implementation: this configuration would be much better off when either included in the command itself (XML metadata for API Platform / Serializer) or at least configurable in `sylius_api` bundle.

Having it defined as tagged services makes it really hard to see how and why does this conversion happens.

Reply:
Or we can get rid of the whole storage class concept by just injecting a map of classes to fields directly to `CommandFieldItemIriToIdentifierDenormalizer` (or creating a new class for that). This way we don't have to create multiple services holding just two strings, there's a dedicated service for managing those instead.

```
class CommandIriArgumentToIdenitiferMap {
  /** @psalm-var array<class-string, string> Command FQCN as key, field as value */
  public function __construct(array $map) {}

  /** 
   * @psalm-var class-string $commandClass
   * @throws CommandNotFound
   */
  public function get(string $commandClass): string {}

  public function has(string $commandClass): bool {}
}
```